### PR TITLE
Topic recognition-Shaivika_48239192_VIT Transformer fot ADNI dataset

### PR DESCRIPTION
**Pull request by shaivikaaaa**
# Alzheimer Image Classification with Vision Transformer (ViT) 

## Overview
This project is an implementation of a Vision Transformer (ViT) for image classification. The code includes components for loading, preprocessing, and training a ViT model on ADNI dataset. Additionally, it allows you to make predictions and provides metrics for model evaluation.

### Files
The project is organized into several files and folders:
- `modules.py`: Contains the implementation of the ViT model components, including patches, patch encoding, and the main classifier.
- `dataset.py`: Contains data loading and preprocessing functions to prepare the image dataset for training.
- `train.py`: Trains the ViT model on the provided dataset and saves the trained model.
- `predict.py`: Uses the trained model to make predictions on single images and visualize the results.

## Results
The application of the Vision Transformer (ViT) model for Alzheimer's disease image classification has yielded promising results. The trained ViT model demonstrates its effectiveness in accurately distinguishing between Alzheimer's disease (AD) and normal control (CN) subjects based on medical images.

### CODE REVIEW

**Comment by aryaman-sh** [comment]
**Difficulty**
- Hard
- Task 6

**Commit messages**
- good

**Code**
- Not adequately commented
- Good structure 

**Readme**
- Image links broken
- Readme good

**Functionality/ Performance**
- Accuracy too high
- Potential dataleak
- Validation loss is always less than train loss
 
**Feedback to implement**
- Fix images in README

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting minimal -1
Data leakage possible -1
Difficulty: Hard
## Commit Log
Meaningful commit messages sometimes -1
Progressive commits used
## Documentation
ReadMe OK, but broken image links -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
